.github/workflows/deploy-mock-api.yml

```diff
@@ -41,10 +41,24 @@ jobs:
 
       - name: Build and Tag Image
         run: |
-          docker build --tag 546642427916.dkr.ecr.us-east-1.amazonaws.com/mock-api-repo:latest ./app
+          docker build --tag 546642427916.dkr.ecr.us-east-1.amazonaws.com/mock-api-repo:latest-${{github.event.inputs.environment || 'test' }} ./app
           aws ecr get-login-password --region us-east-1 | docker login --username AWS --password-stdin 546642427916.dkr.ecr.us-east-1.amazonaws.com/mock-api-repo
-          docker push 546642427916.dkr.ecr.us-east-1.amazonaws.com/mock-api-repo:latest
+          docker push 546642427916.dkr.ecr.us-east-1.amazonaws.com/mock-api-repo:latest-${{github.event.inputs.environment || 'test' }}
 
       - name: Push to ECR
         run: |
           aws ecs update-service --cluster ${{github.event.inputs.environment || 'test' }} --service ${{github.event.inputs.environment || 'test' }}-api-ecs-service --force-new-deployment
+
+  create-tag:
+    needs: [build]
+    name: Create Github Tag
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      # https://github.com/rickstaa/action-create-tag
+      - uses: rickstaa/action-create-tag@v1
+        with:
+          tag: ${{github.event.inputs.environment || 'test' }}
+          message: "from branch ${{github.ref_name}}\nTag created in workflow:\n${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}"
```

Comment:
Could also add the git commit sha instead of the branch

Reply:
I think it's not necessary to use the git commit sha since that's part of the tag metadata.